src/openms/source/ANALYSIS/OPENSWATH/IonMobilityScoring.cpp

```diff
@@ -88,35 +86,67 @@ namespace OpenMS
     double max_int = 0;
     for (Size k = 0; k < im_grid.size(); k++)
     {
+      MobilityPeak1D mobi_peak;
       // In each iteration, the IM value of pr_it should be equal to or
       // larger than the master container. If it is equal, we add the current
       // data point, if it is larger we add zero and advance the counter k.
-      if (pr_it != profile.end() && fabs(pr_it->im - im_grid[k] ) < eps*10)
+      if (pr_it != profile.end() && fabs(pr_it->getMobility() - im_grid[k] ) < eps*10)
       {
-        al_int_values.push_back(pr_it->intensity);
-        al_im_values.push_back(pr_it->im);
+        mobi_peak.setIntensity(pr_it->getIntensity());
+        mobi_peak.setMobility(pr_it->getMobility());
+
         ++pr_it;
       }
       else
       {
-        al_int_values.push_back(0.0);
-        al_im_values.push_back( im_grid[k] );
+        mobi_peak.setIntensity(0.0);
+        mobi_peak.setMobility(im_grid[k]);
       }
       // OPENMS_LOG_DEBUG << "grid position " << im_grid[k] << " profile position " << pr_it->first << std::endl;
 
       // check that we did not advance past
-      if (pr_it != profile.end() && (im_grid[k] - pr_it->im) > eps*10)
+      if (pr_it != profile.end() && (im_grid[k] - pr_it->getMobility()) > eps*10)
       {
-        std::cout << " This should never happen, pr_it has advanced past the master container: " << im_grid[k]  << "  / " <<  pr_it->im  << std::endl;
+        std::cout << " This should never happen, pr_it has advanced past the master container: " << im_grid[k]  << "  / " <<  pr_it->getMobility()  << std::endl;
         throw Exception::OutOfRange(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION);
       }
 
       // collect maxima
-      if (pr_it != profile.end() && pr_it->intensity > max_int)
+      if (pr_it != profile.end() && pr_it->getIntensity() > max_int)
       {
-        max_int = pr_it->intensity;
+        max_int = pr_it->getIntensity();
         max_peak_idx = k;
       }
+
+      aligned_profile.push_back(mobi_peak);
+    }
+  }
+
+  /*
+   @brief Extracts intensity values from a vector of Mobilogram objects
+
+   This function takes a vector of Mobilogram objects and extracts the intensity
+   values from each Mobilogram, storing them in a 2D vector of doubles. The
+   resulting vector of intensity values is stored in the provided output parameter.
+
+   @param mobilograms [in] A const reference to a vector of Mobilogram objects
+                           from which to extract intensity values.
+   @param int_values [out] A reference to a vector of vector of doubles where
+                           the extracted intensity values will be stored. This
+                           vector will be cleared and resized as necessary.
+  */
+  void IonMobilityScoring::extractIntensities(const std::vector< Mobilogram >& mobilograms,
```

Comment:
The `MobilityPeak1D` structure has a [`getIntensity`](https://abibuilder.cs.uni-tuebingen.de/archive/openms/Documentation/nightly/html/classOpenMS_1_1MobilityPeak1D.html#a7aab73a7a5b12869431cc576c87464dc) method, but the `Mobilogram` structure doesn't. The `MSChromatogram` structure has `getFloatDataArrays` and `getIntegerDataArrays`, maybe you're thinking of something similar?

Reply:
I was thinking was there a `get_peaks` method like pyopems however I think this is python specific. https://github.com/OpenMS/OpenMS/blob/188deb57396b98d1f57b00ba30b08b7ecb2fec35/src/pyOpenMS/addons/MSSpectrum.pyx#L6

Since it does not seem like this is a thing in the c++ implementation we can leave this as is

Reply:
yes it is a cython extension that could be added analogously for mobilograms